spec/models/miq_expression/field_spec.rb

```diff
@@ -0,0 +1,96 @@
+require "rails_helper"
+
+RSpec.describe MiqExpression::Field do
+  describe ".parse" do
+    it "can parse the model name" do
+      field = "Vm-name"
+      expect(described_class.parse(field).model).to be(Vm)
+    end
+
+    it "can parse a namespaced model name" do
+      field = "ManageIQ::Providers::CloudManager::Vm-name"
+      expect(described_class.parse(field).model).to be(ManageIQ::Providers::CloudManager::Vm)
+    end
+
+    it "can parse the model name with associations present" do
+      field = "Vm.host-name"
+      expect(described_class.parse(field).model).to be(Vm)
+    end
+
+    it "can parse the column name" do
+      field = "Vm-name"
+      expect(described_class.parse(field).column).to eq("name")
+    end
+
+    it "can parse the column name with associations present" do
+      field = "Vm.host-name"
+      expect(described_class.parse(field).column).to eq("name")
+    end
+
+    it "can parse the column name with pivot table suffix" do
+      field = "Vm-name__pv"
+      expect(described_class.parse(field).column).to eq("name")
+    end
+
+    it "can parse column names with snakecase" do
+      field = "Vm-last_scan_on"
+      expect(described_class.parse(field).column).to eq("last_scan_on")
+    end
+
+    it "can parse the association when there is none present" do
+      field = "Vm-name"
+      expect(described_class.parse(field).association).to be_nil
+    end
+
+    it "can parse the association when there is one present" do
+      field = "Vm.host-name"
+      expect(described_class.parse(field).association).to eq("host")
+    end
+
+    it "will raise a parse error if there are many associations present" do
+      field = "Vm.host.hardware-id"
```

Comment:
It looks like we are trying to support it here: https://github.com/ManageIQ/manageiq/blob/641942f0be16609b2097b5d685b1ed37d564915d/app/models/miq_expression.rb#L1366-L1381, but I don't see how that could work for this use case. Unless I'm very much mistaken (would appreciate another pair of eyes to look over this) I don't think we do, so decided to be explicit about not supporting it here, at least for now.